src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -220,65 +256,91 @@ def test_WINNF_FT_S_FAD_1(self, config_filename):
             self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
                                                dump_file[0])
+            downloaded_file = None
+            if dump_file['recordType'] != 'CoordinationEvent':                
+                downloaded_file = self._sas.DownloadFile(dump_file['url'])
+                datetime.strptime(downloaded_file['startTime'],\
+                                                     '%Y-%m-%dT%H:%M:%SZ')
+                datetime.strptime(downloaded_file['endTime'],\
+                                                     '%Y-%m-%dT%H:%M:%SZ')
             if dump_file['recordType'] ==  'cbsd':
-                cbsd_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])   
+                cbsd_dump_data.append(downloaded_file['recordData'])   
             elif dump_file['recordType'] ==  'esc_sensor':
-                esc_sensor_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+                esc_sensor_dump_data.append(downloaded_file['recordData'])
             elif dump_file['recordType'] ==  'zone':
-                ppa_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+                ppa_dump_data.append(downloaded_file['recordData'])
+            else:
+                self.assertEqual('CoordinationEvent', dump_file['recordType'])
         
         # verify the length of records equal to the inserted ones
         self.assertEqual(len(config['registrationRequests']), len(cbsd_dump_data))
         self.assertEqual(len(config['ppas']), len(ppa_dump_data))
-        self.assertEqual(len(config['esc_sensors']), len(esc_sensor_dump_data))
-        
-        # verify the schema of record and that the injected ppas exist in the dump files
-        for index, ppa in enumerate(config['ppas']):
-            ppa_record = [record for record in ppa_dump_data if record['id'].split("/")[-1] == ppa_ids[index]][0]
+        self.assertEqual(len(config['escSensors']), len(esc_sensor_dump_data))
+        # verify the schema of record and first two parts of PPA record Id  
+        for ppa_record in ppa_dump_data:
+            self.assertContainsRequiredFields("zoneData.schema.json", ppa_record)              
+            self.assertEqual(ppa_record['id'].split("/")[0], 'zone')
+            self.assertEqual(ppa_record['id'].split("/")[1], self._sas._sas_admin)
             del ppa_record['id']
-            self.assertContainsRequiredFields("zoneData.schema.json", ppa_record)
-            self.assertDictEqual(ppa, ppa_record)
+        # verify that the injected ppas exist in the dump files
+        for index, ppa in enumerate(config['ppas']):
+            del ppa['id']
+            exist_in_dump = False
+            for ppa_record in ppa_dump_data:
+                shared_items = set(ppa_record.items()) & set(ppa.items())               
+                if  len(shared_items) == len(ppa):
```

Comment:
Shouldn't you be looking for len(shared_items) to be equal to 1? `len(ppa)` will yield the number of keys in the dict `ppa`.